### PR TITLE
embed_id skip checks validity if value equals ignore label.

### PR DESCRIPTION
**Pull request by fukatani**
This PR is for skipping checks validity if value equals `ignore_label` (#1351 ).
`embed_id ` will not raise error value is "invalid" but equals `ignore_label`.

### CODE REVIEW

File: chainer/functions/connection/embed_id.py [hunk 1]
```diff
@@ -27,14 +27,16 @@ def check_type_forward(self, in_types):
...
 
+        xp = cuda.get_array_module(*inputs)
         if chainer.is_debug():
-            if not ((0 <= x).all() and
-                    (x < len(W)).all()):
+            valid_x = xp.logical_and(0 <= x, x < len(W))
+            if self.ignore_label is not None:
+                valid_x = xp.logical_or(valid_x, x == self.ignore_label)
+            if not valid_x.all():
                 msg = 'Each `x` value need to satisfy `0 <= x < len(W)`'
```
Comment (delta2323) [inline chainer/functions/connection/embed_id.py]: Could you change the error message as well?

**Comment by fukatani** [comment]
Thanks! I try to the error massage to be more precise.

File: tests/chainer_tests/links_tests/connection_tests/test_embed_id.py [hunk 2]
```diff
@@ -69,14 +69,15 @@ def test_backward_gpu(self):
 
 
 @testing.parameterize(
-    {'t_value': -1, 'valid': False},
-    {'t_value': 3,  'valid': False},
-    {'t_value': 0,  'valid': True},
+    {'t_value': -1, 'valid': False, 'ignore_label': None},
+    {'t_value': 3,  'valid': False, 'ignore_label': None},
+    {'t_value': 0,  'valid': True,  'ignore_label': None},
+    {'t_value': -1, 'valid': True,  'ignore_label': -1},
```
Comment (delta2323) [inline tests/chainer_tests/links_tests/connection_tests/test_embed_id.py]: Could you add test cases where `t_value = 0, 3` and `ignore_label = -1`?

**Comment by delta2323** [comment]
LGTM except the comment.

**Comment by delta2323** [comment]
Thank you for your contribution!

**Pull request merged**